### PR TITLE
🔄 Displays sync history in dashboard

### LINKED ISSUE #85: Display sync history on Dashboard
As an Admin
When I view the dashboard
I can a list of entries where
* each entry represents an attempt to sync Chowda with Sony Ci
* each entry shows the date/time of the attempt, and whether the attempt succeeded or failed
* entries are ordered by date/time in descending order
* entries are limited to the most recent 10 attempt.
In order to determine whether Chowda has recently been fully sync'd with Sony Ci

### LINKED ISSUE #87: BUG: Dashboard should not error if no Sync has been run yet
### To reproduce:
1. Start Chowda with no `IngestFlow` having been run yet.
2. Go to Dashboard page at `/admin`

### Expected result
See a message indicating that no Sync has yet been run

### Actual result
Error

### PR DESCRIPTION
**Pull request by afred**
# Dashboard Sync History
Adds widget to display list of recent sync events

Closes #85
Closes #87

### CODE REVIEW

**Comment by afred** [comment]
image

File: chowda/views.py [hunk 1]
```diff
@@ -128,14 +129,19 @@ class ClamsEventView(ModelView):
 
 
 class DashboardView(CustomView):
-    def sony_ci_last_sync(self):
-        # TODO: replace with actual "last sync" time
-        return datetime.now()
+    def sync_history(self) -> Dict[str, Any]:
+        return [
+            {'created_at': sync_run.created_at, 'successful': sync_run.successful}
+            for sync_run in list(Flow('IngestFlow'))
```
Comment (mrharpo) [inline chowda/views.py:135]: When running in production, this errored with: `metaflow.exception.MetaflowNotFound` and crashed the server.

We should:
- [ ] Protect this block from not finding a flow
- [ ] Grab the correct results from `namespace('production:ingestflow-0-sdjk')`

**Review by mrharpo — CHANGES_REQUESTED** [review]
👍 Just need to protect from error cases.

File: templates/dashboard.html [hunk 2]
```diff
@@ -9,37 +9,60 @@ <h1>Dashboard</h1>
...
+                                        <tr>
+                                            <th>Date / Time</th>
+                                            <th>Status</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        {% for sync in sync_history %}
+                                        <tr>
+                                            <td>
+                                                {{ sync['created_at'].strftime('%Y-%m-%d %I:%M:%S %p') }}
```
Comment (mrharpo) [inline templates/dashboard.html:34]: Can we change this to an iso8601 string for the frontend to parse?

```py
sony_ci_last_sync.isoformat()
```

Comment (afred) [reply templates/dashboard.html:34]: Can you elaborate on what you're looking for here?
```
sync['created_at'].strftime('%Y-%m-%d %I:%M:%S %p')
=> 2023-07-18 11:53:31 AM

sync['created_at'].isoformat()
=> 2023-07-18T11:53:31.965000

sync['created_at']
=> 2023-07-18 11:53:31.965000

```
The first one is more user friendly, and this template is the last stop before the front end display.

**Review by mrharpo — APPROVED** [review]
👍

**Pull request merged**